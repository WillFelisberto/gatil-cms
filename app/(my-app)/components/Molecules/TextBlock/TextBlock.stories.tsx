import type { Meta, StoryObj } from '@storybook/react';

import type { BlockProps } from '@/app/(my-app)/types/block-props';

import { TextBlock } from './TextBlock';

const defaultProps: BlockProps<'textBlock'> = {
  content: {
    root: {
      type: 'root',
      format: '',
      indent: 0,
      version: 1,
      direction: 'ltr',
      children: [
        {
          type: 'paragraph',
          version: 1,
          indent: 0,
          format: '',
          direction: 'ltr',
          children: [
            {
              type: 'text',
              version: 1,
              text: 'Homepage',
              detail: 0,
              format: 0,
              mode: 'normal',
              style: ''
            }
          ]
        }
      ]
    }
  },
  blockType: 'textBlock'
};

const meta: Meta<typeof TextBlock> = {
  title: 'Molecules/TextBlock',
  component: TextBlock,
  tags: ['autodocs'],
  args: defaultProps
};

export default meta;

type Story = StoryObj<typeof TextBlock>;

export const Default: Story = {};
