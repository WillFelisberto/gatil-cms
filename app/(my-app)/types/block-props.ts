import type { Homepage } from '@/payload-types';

type Layout = Homepage['layout'];

export type BlockProps<K extends Layout[number]['blockType']> = Extract<
  Layout[number],
  { blockType: K }
>;
