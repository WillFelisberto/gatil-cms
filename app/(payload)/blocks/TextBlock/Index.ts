import { Block } from 'payload';

export const TextBlock: Block = {
  slug: 'textBlock',
  labels: { singular: 'Texto', plural: 'Textos' },
  fields: [
    {
      name: 'content',
      label: 'Texto',
      type: 'richText',
      required: true
    }
  ]
};
