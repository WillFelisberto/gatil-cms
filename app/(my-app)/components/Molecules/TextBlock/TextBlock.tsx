import { BlockProps } from '@/app/(my-app)/types/block-props';

import { RichTextComponent } from '../../Atoms/RichText';

export const TextBlock = (props: BlockProps<'textBlock'>) => {
  const { content } = props;
  return (
    <div className="text-gray-600 mb-4 leading-relaxed">
      <RichTextComponent lexicalData={content} />
    </div>
  );
};
