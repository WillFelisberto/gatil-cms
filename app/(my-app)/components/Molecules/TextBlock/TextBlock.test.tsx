import { BlockProps } from '@/app/(my-app)/types/block-props';
import { render, screen } from '@/tests/test-utils';

import { TextBlock } from './TextBlock';

const defaultProps: BlockProps<'textBlock'> = {
  content: {
    root: {
      type: 'root',
      format: '',
      indent: 0,
      version: 1,
      direction: 'ltr',
      children: [
        {
          type: 'paragraph',
          version: 1,
          indent: 0,
          format: '',
          direction: 'ltr',
          children: [
            {
              type: 'text',
              version: 1,
              text: 'Homepage',
              detail: 0,
              format: 0,
              mode: 'normal',
              style: ''
            }
          ]
        }
      ]
    }
  },
  blockType: 'textBlock'
};

describe('<TextBlock />', () => {
  it('Should renders the rich text content', () => {
    const { container } = render(<TextBlock {...defaultProps} />);
    expect(screen.getByText('Homepage')).toBeInTheDocument();

    expect(container).toMatchSnapshot();
  });
});
