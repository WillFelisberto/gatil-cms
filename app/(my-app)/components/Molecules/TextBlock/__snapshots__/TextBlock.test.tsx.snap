// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TextBlock /> Should renders the rich text content 1`] = `
<div>
  <div
    class="text-gray-600 mb-4 leading-relaxed"
  >
    <div
      class="payload-richtext"
    >
      <p>
        Homepage
      </p>
    </div>
  </div>
</div>
`;
